src/script/standard.cpp

```diff
@@ -159,13 +159,15 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::v
             else if (opcode2 == OP_SMALLINTEGER)
             {   // Single-byte small integer pushed onto vSolutions
                 if (opcode1 == OP_0 ||
-                    (opcode1 >= OP_1 && opcode1 <= OP_16))
-                {
+                    (opcode1 >= OP_1 && opcode1 <= OP_16)) {
                     char n = (char)CScript::DecodeOP_N(opcode1);
                     vSolutionsRet.push_back(valtype(1, n));
-                }
-                else
+                } else if (opcode1 == 0x01 && CScriptNum(vch1, true).getint() > 16 && CScriptNum(vch1, true).getint() <= 20) {
```

Comment:
Do we handle scriptnum_error here?

Reply:
it should be a single byte here, yes? I'm not as knowledgeable on Bitcoin script numerics.

Reply:
Hmm? I'm referring to the fact that CScriptNum will throw scriptnum_error here if vch1 is {0x80}.